app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -1130,7 +1130,10 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
     if (event.getAction() == KeyEvent.ACTION_DOWN) {
       switch (keyCode) {
         case KeyEvent.KEYCODE_BACK:
-          if (getCurrentWebView().canGoBack()) {
+          if (tabSwitcherRoot.getVisibility() == View.VISIBLE) {
+            selectTab(webViewList.size() - 1);
```

Comment:
This will hide the tab switcher but will always take me to the last tab in the list. Instead it should take me to the last tab viewed before the tab switcher was opened. Google Chrome handles it in the same way.

Reply:
@abdulwd  Now onBackpressed, last view tab will open. Please review it.